libcloud/storage/base.py

```diff
@@ -639,13 +624,30 @@ def _upload_object(self, object_name, content_type, request_path,
         if not response.success():
             response.parse_error()
 
-        if upload_func:
-            upload_func(**upload_func_kwargs)
-
         return {'response': response,
                 'bytes_transferred': stream_length,
                 'data_hash': stream_hash}
 
+    def _determine_content_type(self, content_type, object_name,
+                                file_path=None):
+        if not content_type:
```

Comment:
I would probably inverse that logic and do if content type and return early to avoid one level of nesting, but that's just a personal style preference.

Reply:
Cleaned up the function in 498ac88.